nemo_curator/modules/fuzzy_dedup.py

```diff
@@ -508,6 +657,7 @@ def _get_output_map_based_on_str_bytes(
         """
         Add output_partition_id to buckets_ddf
         """
+        documents_df = documents_df.copy()
```

Comment:
Found the spot causing the `_text_bytes` column side effect. Here `copy` should not have any significant perf implications since it just copies the task graph not the actual data itself.